### PR TITLE
fix string return type

### PR DESCRIPTION
**Pull request by oliviazhang0809**

### CODE REVIEW

File: codegen/method.go [hunk 1]
```diff
@@ -862,19 +865,36 @@ func (ms *MethodSpec) setTypeConverters(
...
-
 	respConverter := &TypeConverter{
 		LineBuilder: LineBuilder{},
 		Helper:      h,
 	}
 
-	err = respConverter.GenStructConverter(downstreamRespFields, respFields, respTransforms)
-	if err != nil {
-		return err
+	respConverter.append("func convert", pascalCase(ms.Name), "ClientResponse(in ", downstreamMethod.ResponseType, ") ", ms.ResponseType, "{")
```
Comment (Raynos) [inline codegen/method.go]: we dont have a lint rule yet for 80 characters but we try to stick to it, can add some linebreaks here for readability.

File: codegen/templates/endpoint.tmpl [hunk 2]
```diff
@@ -171,6 +171,8 @@ func (handler *{{$handlerName}}) HandleRequest(
 
 	{{if eq .ResponseType "" -}}
 	res.WriteJSONBytes({{.OKStatusCode.Code}}, cliRespHeaders, nil)
+	{{- else if eq .ResponseType "string" -}}
+	res.WriteJSONBytes({{.OKStatusCode.Code}}, cliRespHeaders, []byte(response))
```
Comment (Raynos) [inline codegen/templates/endpoint.tmpl]: I don't think this is valid json ? `"\"" + response + "\""` ?

Comment (oliviazhang0809) [reply codegen/templates/endpoint.tmpl]: the response is a string itself. 
check out `build/endpoints/bar/bar_bar_method_helloworld.go`

File: test/endpoints/bar/bar_hello_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+	assert.Equal(t, "200 OK", res.Status)
+	assert.Equal(t, 1, counter)
+
+	respBytes, err := ioutil.ReadAll(res.Body)
+	if !assert.NoError(t, err, "got http resp error") {
+		return
+	}
+
+	assert.Equal(t, string(respBytes), "hello")
```
Comment (Raynos) [inline test/endpoints/bar/bar_hello_test.go]: this is still not valid JSON. We should expect `"hello"` here. 

Also need to test what happens if send `"\"hello\""` and if strings are properly encoded and stuff.

File: test/endpoints/bar/bar_hello_test.go [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	if !assert.NoError(t, err, "got bootstrap err") {
+		return
+	}
+	defer gateway.Close()
+
+	gateway.HTTPBackends()["bar"].HandleFunc(
+		"GET", "/bar/hello",
+		func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(200)
+			if _, err := w.Write([]byte(`"hello"`)); err != nil {
```
Comment (Raynos) [inline test/endpoints/bar/bar_hello_test.go]: Try doing `json.Marshal("I am\"string\"with\"nestedquotes")` or something and double check that the `json.Unmarshal()` from `res.Body` contains the same string.

**Pull request merged**